lib/Dialect/Moore/MooreOps.cpp

```diff
@@ -1637,6 +1637,65 @@ VTableLoadMethodOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
   return success();
 }
 
+LogicalResult VTableOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  Operation *self = getOperation();
+
+  // sym_name must end with ::@vtable and root must be a ClassDeclOp
+  SymbolRefAttr name = getSymNameAttr();
+  if (!name)
+    return emitOpError("requires 'sym_name' SymbolRefAttr");
+
+  // Last component must be "vtable".
+  if (name.getNestedReferences().empty() ||
+      name.getNestedReferences().back().getValue() != "vtable")
+    return emitOpError() << "sym_name must end with '::@vtable', got " << name;
```

Comment:
Can we drop this requirement? The `VTableOp` could just point to the `ClassDeclOp` directly, and that's it 🙂